### DISCUSSION TITLE
How to read zarr?

### DISCUSSION THREAD

**Discussion by Break00** [cogeotiff/rio-tiler#511] [Q&A] [answered]
import xarray as xr
import zarr
from rio_tiler.io import COGReader

# read zarr
ds = xr.open_zarr(r'D:\work\data\test.zarr')

# I want to read zarr by COGReader, but  Error !!!
with COGReader(None,dataset=ds) as image:
    img = image.read()

#################################
Traceback (most recent call last):
  File "D:\python\zarr\test.py", line 43, in <module>
    with COGReader(None,dataset=ds) as image:
  File "<attrs generated init rio_tiler.io.cogeo.COGReader>", line 16, in __init__
  File "C:\Users\dell\AppData\Roaming\Python\Python39\site-packages\rio_tiler\io\cogeo.py", line 135, in __attrs_post_init__
    self.bounds = tuple(self.dataset.bounds)
  File "C:\Users\dell\AppData\Roaming\Python\Python39\site-packages\xarray\core\common.py", line 256, in __getattr__
    raise AttributeError(
AttributeError: 'Dataset' object has no attribute 'bounds'

**Comment by vincentsarago** [comment, marked as answer]
Rio-tiler only accept rasterio object not fsspec ones.

Have a look on the cogeotiff/rio-tiler-fs project 

Also there is no guarantee that rio-tiler will fully support Zarr dataset because GDAL support is somehow complicated (it splits the dataset in subdataset only for dataset with more than 2 dim)

See #453

Edit: oh I miss read what you were doing, but almost the same answer rio-tiler only accept rasterio object not xarray ones

I'm planning to support Zarr at some point but the specification and GDAL native support makes it non trivial

**Reply by Break00** [reply]
Thanks, the main purpose is to use rio-tiler to read zarr from the cloud to implement dynamic tile service. Looking forward to your plan!
